### DISCUSSION TITLE
Date Formatting

### DISCUSSION THREAD

**Discussion by EventHorizonsEngineering** [kimai/kimai#4837] [General] [answered]
I just started testing the trial version on kimai cloud.  So far I am vary happy with how it works for my small business.

    I am an English user in the USA who uses 24-hour time.   I use Firefox (115.10.0esr 64-bit) on Rocky 9 (RHEL) Linux. 

    Since the plain "English" option does not support 24-hour time, I at first selected "English(Indonesia)" because it supports 24-hour time and the date is MM/DD/YYYY.  I find the leading zeros and 4-digit year to more readable and pleasant. 

    The other day in "My Times" it started showing the date as DD/MM/YYYY.  I checked my settings but did not change them and the displayed dates went back to normal. (At least I think they did!)

    Today it started doing the same thing. 

    If I choose plain "English" then the date shows at mm/dd/yy (ugh!) but I have tried multiple alternate versions of English that claim to be MM/DD/YYYY, yet the date is still shown as DD/MM/YYYY.

    I noticed that there was a "Cloud Update 106" on 05/03/2024.  
    Is there an option or a setting that I am not seeing to resolve this or should I look into reporting it as a bug?

    Thanks!

**Comment by kevinpapst** [comment]
There are only two spanish locales (es_PA, es_PR) that use MM/DD/YYYY, both use 12 hour format.

As long as there is no free choice for date and time format, you have to use one of the provided locales with ICU settings.

**Reply by EventHorizonsEngineering** [reply]
My apologies!  I was not clear as to the problem I am experiencing. 

Under "Preferences" and "Date, Time and Currency Format"  I can choose from a long list of options.

If I select "English" then the date is shown as 5/4/24 and the time is shown as AM/PM.  This is not my display preference.
If I select "English(Ireland)" then the date is supposed to be "05/04/2024" (MM/DD/YYYY) and the time is 24-hour format (also, the currency indicator is at the front and the comma is used for thousands)

This worked at first, but now the dates in "My Times" as well as other places show as "04/05/2024" (DD/MM/YYYY).  I tried "English(Indonesia)", "English(United Arab Emirates)", "English(Germany)" and several others with the same effect.    I have not tried all possible combinations, but all the English locales I tried that are supposed to be Month-Day-Year are showing as Day-Month-Year

The "...kimai.cloud/en/help/locales" shows them correctly but in "My Times" they are displayed incorrectly.

**Reply by kevinpapst** [reply]
Can you show me a screenshot of your user preferences. My times does not use the format you configured?

**Reply by EventHorizonsEngineering** [reply, marked as answer]
Screenshot at the bottom, but it's no longer relevant. 

I loaded a fresh version of the "locales" page to also take a screen shot of it and discovered that the date formatting for many of the English versions have changed.   

Now every English version that supports 24-hour time has Date-Month-Year except for "English (Burundi)" and two that were "Year-Month-Day".    

That does explain why the formatting that was shown on the locales page differed from what I was getting when I selected a particular locale.    I was using a version of the locales page that was cached from when I first bumped into this issue a few days ago!

I'm assuming that the formatting is derived from somewhere (such as "IntlDateFormatter" or similar) and they are not individually defined in some kind of template file.

"English (Burundi)" has a two-digit year and no leading zeros, which is not ideal but usable.  "My Times" now matches what I see on the locales page, which was my largest initial concern.

Ideally there would be an option to enable 24-hour time but I understand the effort possibly required for that.

![Screenshot from 2024-05-06 12-14-26](https://github.com/kimai/kimai/assets/168922307/9431abc7-514c-4bd7-9d20-60320078b23c)

**Reply by kevinpapst** [reply]
At some point there will be configurable settings for formats. 
For now locales are simpler to support.
And yes, the settings are generated from the IntLDateFormatter. 
Just tried to regenerate and there was no change.

**Reply by EventHorizonsEngineering** [reply]
I appreciate your time!  Thanks!
